### PR TITLE
Support for additional arguments during build

### PR DESCRIPTION
**Pull request by juliusloman**
Added support for passing additional arbitrary arguments to docker build command. For example --build-arg.  I use it for passing HTTP proxy information to docker daemon during build.

### CODE REVIEW

File: src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java [hunk 1]
```diff
@@ -74,8 +74,10 @@
     private boolean createFingerprint = true;
     private boolean skipTagLatest;
 
+    private String buildAdditionalArgs = "";
+
     @Deprecated
-    public DockerBuilder(String repoName, String repoTag, boolean skipPush, boolean noCache, boolean forcePull, boolean skipBuild, boolean skipDecorate, boolean skipTagLatest, String dockerfilePath) {
```
Comment (carlossg) [inline src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java]: could you keep the previous constructor for binary compatibility ? it should be enough with the @DataBoundSetter

**Comment by juliusloman** [comment]
Sure. Sorry my fault :-) I've restored the original one.

**Comment by carlossg** [comment]
sorry, what I meant is that you don't need to touch the constructors, the `@DataBoundSetter` annotation will make the field be injected.
Other than that LGTM, If you remove the added constructor and squash the commits I'll merge

**Comment by juliusloman** [comment]
Removed the new constructor.

**Comment by carlossg** [comment]
can you squash the commits into one?

**Comment by juliusloman** [comment]
Squashed into single commit.

**Comment by carlossg** [comment]
Thanks!

**Pull request merged**